presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SymbolMapper.java

```diff
@@ -68,74 +74,98 @@ public Expression rewriteSymbolReference(SymbolReference node, Void context, Exp
         }, value);
     }
 
-    public AggregationNode map(AggregationNode node, PlanNode source)
+    public <T extends PlanNode> T map(PlanNode planNode, PlanNodeIdAllocator idAllocator, RecursionStrategy recursionStrategy)
```

Comment:
My main goal is to keep the non-recursive interface as simple as possible, even at the expense of a slightly more tedious implementation behind the interface.  I think the approach in the PR is okay, though, and I understand that you might weigh these goals differently.